Expand and consolidate networking docs (#68051)
elastic/elasticsearch: docs/reference/modules/network/tracers.asciidoc

==== Request tracing

It is possible to trace individual requests made on the HTTP and transport
layers. Be warned that this tracing can generate extremely high log volumes
which may destabilise your cluster. Do not enable request tracing on busy or

> important clusters.

Reviewer: GitHub won't handle that change as a suggestion that can be easily committed. Given the gravity of the message, I'd consider making it a WARNING admonition.

[[http-rest-request-tracer]]
===== REST request tracer

> The HTTP layer has a dedicated tracer logger which, when activated, logs
> incoming requests and the corresponding outgoing responses. The log can be
> dynamically activated by setting the level of the
> `org.elasticsearch.http.HttpTracer` logger to `TRACE`:

Reviewer: ```suggestion
The HTTP layer has a dedicated tracer that logs
incoming requests and the corresponding outgoing responses. Activate
the tracer by setting the level of the
`org.elasticsearch.http.HttpTracer` logger to `TRACE`:
```

[source,console]
--------------------------------------------------
PUT _cluster/settings
{
   "transient" : {
      "logger.org.elasticsearch.http.HttpTracer" : "TRACE"
   }
}
--------------------------------------------------

You can also control which URIs will be traced, using a set of include and
exclude wildcard patterns. By default every request will be traced.

[source,console]
--------------------------------------------------
PUT _cluster/settings
{
   "transient" : {
      "http.tracer.include" : "*",
      "http.tracer.exclude" : ""
   }
}
--------------------------------------------------

[[transport-tracer]]
===== Transport tracer

> The transport layer has a dedicated tracer logger which, when activated, logs
> incoming and outgoing requests and responses. The log can be dynamically
> activated by setting the level of the
> `org.elasticsearch.transport.TransportService.tracer` logger to `TRACE`:

Reviewer: ```suggestion
The transport layer has a dedicated tracer that logs
incoming and outgoing requests and responses. Activate the tracer 
by setting the level of the
`org.elasticsearch.transport.TransportService.tracer` logger to `TRACE`:
```
